lib/Dancer2/Core/App.pm

```diff
@@ -889,7 +889,11 @@ sub send_file {
                 [ $res->status, $res->headers_to_array, $fh ]
             );
         };
+
+        Scalar::Util::weaken( my $weak_self = $self );
+
         $response = Dancer2::Core::Response::Delayed->new(
+            error_cb => sub { $weak_self->logger_engine->log(@_) },
```

Comment:
I'll amend it to contain the fix. Thanks! Unfortunately the development server's failure is kind of weird. I couldn't write a proper test for it.